juju/juju
# Add support for lxd snap socket, drop juju wrapper in snap

## Description of change

Adds support for snap socket of LXD, and prefers it. Drops the wrapper in snap that previously allowed this support.

## QA steps

sudo snap install lxd
juju bootstrap lxd
Juju should bootstrap using lxd socket

sudo apt install lxd (now 2 lxd's are installed)
juju bootstrap lxd
Juju should still bootstrap using lxd socket

sudo snap remove lxd (now only deb lxd installed)
juju bootstrap lxd
Juju should bootstrap using debian socket

sudo apt remove lxd (no lxd installed)
juju bootstrap lxd
Juju should prompt to install lxd snap

## Bug reference

Fixes LP [#1743719](https://bugs.launchpad.net/juju/+bug/1743719)
